site/content/contribute/webapp/unit-testing.md

```diff
@@ -14,43 +14,37 @@ The last required piece of building a webapp component is to test it. That can b
 
 Below is a brief guide on how to do component testing:
 
-1. Match snapshot using default or expected props. Note that while the snapshot is convenient, we require not to rely solely on this for every test case as this is easily overlooked by initiating `jest -updateSnapshot` without carefully inspecting the change.
+1. Match snapshot using default or expected props. Note that while the snapshot is convenient, we shouldn't rely solely on this for every test case as changes can be easily overlooked when using `jest -updateSnapshot` to update multiple snapshots at once.
     ```javascript
     const baseProps = {
-        active        onSubmit: jest.fn(),
-        update    };
+        active: true,
+        onSubmit: jest.fn(),
+    };
 
     test('should match snapshot, not send email notifications', () => {
         const wrapper = shallow(<EmailNotificationSetting {...baseProps}/>);
 
-        // Use "toMatchInlineSnapshot" whenever possible when the snapshot consists of several lines of code only
-        // It creates an easier to read snapshot, inline with the test file.
-        expect(wrapper).toMatchInlineSnapshot();
-
-        // Save snapshot particularly when component has other render function like "renderOption"
-        // It creates a small snapshot of that particular render function instead of the entire component
-        expect(wrapper.instance().renderOption()).toMatchInlineSnapshot();
-
-        // Only use "toMatchSnapshot" whenever above options are not possible.
-        // Limit the use to one (1) snapshot only.
-        // Save snapshot if it generates an easy to inspect and identifiable HTML or components that can easily verify future change.
         expect(wrapper).toMatchSnapshot();
+
+        // Save a snapshot of part of a component when it has another render function like "renderOption".
+        // This creates a small snapshot of that particular render function's result instead of the entire component.
+        expect(wrapper.instance().renderOption()).toMatchSnapshot();
```

Comment:
this would only work with class-based components .. should we mention that here?

Reply:
I don't think we need to mention it here. If it's a functional component, I'd just have that method into another component which could be tested independently, so this example isn't really relevant for them